falcon/api.py

```diff
@@ -103,6 +103,10 @@ def process_response(self, req, resp, resource, req_succeeded)
             to use in lieu of the default engine.
             See also: :ref:`Routing <routing>`.
 
+        router_search_method (str, optional): The name of a custom router method
```

Comment:
> There's the possibility that existing custom router implementations will already use whatever method name we choose for some other functionality.

Good point. I don't think there is any way around potentially breaking a custom router if we introduce another method name. Making that name configurable provides a way to mitigate the break, but doesn't avoid it in the first place. At this point a think we have the following options:

**a)** Introduce a new method in the next minor point release, which may break some custom routers out there.

**b)** Don't introduce a new method, but wait until 2.0 to change the `find()` method to take a `Request` object instead of a path

**c)** Don't introduce a new method, but add a `req` param to `find()` for the next minor point release. Then remove the path param in 2.0 as a breaking change. When a custom router is passed in, check the argspec and call in one of two ways, accordingly, to be backwards-compatible.

I'm keen to not do (a) because that goes against semver. (c) means passing redundant params, but it gets this functionality in people's hands sooner than (b), and can be cleaned up with the 2.0 release anyway. 

What do you think? I would be happy with either (b) or (c).